pkgs/applications/editors/lapce/_maintainers/plugins_update.nu

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i nu -p nushell
+
+cd $env.FILE_PWD
+cd .. # exit _maintainers folder
+
+$env.NIX_CONFIG = "experimental-features = nix-command flakes\n"
+
+let plugins_dir = "plugins"
+let index_path = $"($plugins_dir)/.#index.json"
+
+mkdir $plugins_dir
+
+if not ($index_path | path exists) {
+    mut plugins = []
+    mut offset = 0
```

Comment:
Instead of using a loop and mutable state you can grab the plugins using the `generate` function

```
generate {|offset=0|
    let limit = 50
    let plugins = (http get $"https://plugins.lapce.dev/api/v1/plugins?offset=($offset)&limit=($limit)").plugins

    if ($plugins|is-not-empty) {
        {
            out: $plugins
            next: ($offset + ($plugins|length))
        }
    }
}
| flatten
```